engines/bops_core/app/components/bops_core/header_bar_component.rb

```diff
@@ -0,0 +1,117 @@
+    end
+
+    private
+
+    attr_reader :left_items, :right_items, :toggle
+
+    def container_classes
+      "bops-header-bar"
+    end
+
+    def container_data_attributes
+      return {} unless toggle_options
+
+      {
+        controller: "toggle",
+        toggle_class_name_value: toggle_options[:class_name],
+        toggle_condensed_text_value: toggle_options[:condensed_text],
+        toggle_expanded_text_value: toggle_options[:expanded_text]
+      }
+    end
+
+    def inner_classes = "bops-header-bar__inner"
+    def left_stack_classes = "bops-header-bar__left"
+    def right_stack_classes = "bops-header-bar__right"
+    def divider_classes = "bops-header-bar__divider"
+    def base_link_classes = "govuk-link"
+
+    def render_left_items
+      helpers.content_tag(:ul, class: left_stack_classes, role: "list") do
+        items = left_items.map { |item| render_left_text_item(item) }
+        items << render_left_toggle_item if toggle_options
+        safe_join(items)
+      end
+    end
+
+    def render_left_text_item(item)
+      helpers.content_tag(:li, class: "bops-header-bar__item") do
+        helpers.tag.span(left_item_text(item), class: left_item_class(item))
+      end
+    end
+
+    def render_left_toggle_item
+      helpers.content_tag(:li, class: "bops-header-bar__item--toggle") do
+        render_toggle_button
+      end
+    end
+
+    def left_item_text(item)
+      item[:text].to_s
+    end
+
+    def left_item_class(item)
+      class_names(
+        "bops-header-bar__text",
+        ("govuk-!-font-weight-bold" if item[:bold].present?),
+        item[:class].presence
+      )
+    end
+
+    def render_toggle_button
+      return unless toggle_options
+
+      helpers.button_tag(
+        toggle_options[:condensed_text],
+        type: "button",
+        class: class_names("button-as-link", "govuk-link", "bops-header-bar__toggle-button", toggle_options[:button_class]),
+        data: {toggle_target: "button", action: "toggle#click"},
+        "aria-expanded": "false"
+      )
+    end
+
+    def render_toggle_content
+      return unless toggle_options
+
+      helpers.tag.div(
+        toggle_options[:content],
+        class: class_names("bops-header-bar__toggle-panel", toggle_options[:class_name], toggle_options[:content_class]),
+        data: {toggle_target: "content"},
+        "aria-live": "polite"
+      )
+    end
+
+    def link_classes(item)
+      class_names(base_link_classes, item[:class].presence)
+    end
+
+    def right_is_link?(item)
+      item[:href].present?
+    end
+
+    def toggle_options
+      @toggle_options ||= begin
+        return nil if toggle.blank?
+
+        content = toggle[:content]
+        return nil if content.blank?
+
+        {
+          class_name: toggle[:class_name].presence || "govuk-!-display-none",
+          condensed_text: toggle[:condensed_text].presence || "Show details",
+          expanded_text: toggle[:expanded_text].presence || "Hide details",
```

Comment:
This will all be falsy even without `presence`:

```suggestion
          class_name: toggle[:class_name] || "govuk-!-display-none",
          condensed_text: toggle[:condensed_text] || "Show details",
          expanded_text: toggle[:expanded_text] || "Hide details",
```